src/Ros_mpGetRobotCalibrationData.c

```diff
@@ -0,0 +1,160 @@
+// Ros_mpGetRobotCalibrationData.c
+
+// SPDX-FileCopyrightText: 2023, Yaskawa America, Inc.
+// SPDX-FileCopyrightText: 2023, Delft University of Technology
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "MotoROS.h"
+
+#define FILENAME_RBCALIB_DAT    "RBCALIB.DAT"
+#define PATH_TO_RBCALIB_DAT     "MPRAM1:0\\RBCALIB.DAT"
+
+
+LONG Ros_mpGetRobotCalibrationData(ULONG file_no, MP_RB_CALIB_DATA *rData)
+{
+#if defined (YRC1000) || defined (YRC1000u)
+    // on these controllers we can forward to mpGetRobotCalibrationData(..)
+    return mpGetRobotCalibrationData(file_no, rData);
+
+#elif defined (DX200) || defined (FS100)
+    //The download and parsing of the DAT file is expensive. I only want to
+    //perform it once. So let's cache the data in an array.
+    //
+    //This is an array of POINTERS. We'll only allocate a slot if that calibration
+    //file is configured.
+    static MP_RB_CALIB_DATA* calibrationFiles[MAX_ROBOT_CALIBRATION_FILES];
+    
+    if (file_no == 0) //first time in, let's parse the whole file
```

Comment:
Are all 'files' stored in the single `RBCALIB.DAT` file?

Reply:
Yes. Example DAT file: 
[RBCALIB.DAT.TXT](https://github.com/Yaskawa-Global/motoros2/files/11969549/RBCALIB.DAT.TXT)

Reply:
hm. Would it perhaps not be more robust/defensive to check for an entry in `calibrationFiles` instead of relying on the caller starting at `0`?

Reply:
This behavior has been modified in [fe22ac3](https://github.com/Yaskawa-Global/motoros2/pull/49/commits/fe22ac3506855c19873064975acddb3856981963). It now gets parsed during initialization.